frontend/packages/dev-console/src/utils/pipeline-augment.ts

```diff
@@ -143,3 +187,31 @@ export const augmentRunsToData = (
   }
   return data;
 };
+
+export const getTaskStatus = (pipelinerun: PipelineRun): TaskStatus => {
+  const taskStatus = { Succeeded: 0, Running: 0, Failed: 0, Notstarted: 0, FailedToStart: 0 };
+  if (pipelinerun && pipelinerun.status && pipelinerun.status.taskRuns) {
```

Comment:
Looks like the status of the pipelinerun doesn't always represent the full set of tasks.
For example in your sample, there are 5 tasks:
![image](https://user-images.githubusercontent.com/14068621/60126841-d81bd980-975c-11e9-8403-bfc9b967109a.png)

Yet when this is re-run, it never shows those that have not yet started. Here it says 1 is running, but doesn't show that 4 have not yet started:
![image](https://user-images.githubusercontent.com/14068621/60126879-ef5ac700-975c-11e9-820b-75b387b2d349.png)

As the it progresses, it does flash not started for the next task before showing the status of the next set of tasks to run:
![image](https://user-images.githubusercontent.com/14068621/60126944-12857680-975d-11e9-95ac-16a1901fd34a.png)

Finally once complete it shows 5 succeeded.

Reply:
The problem with this is that the `PipelineRun` data doesn't have any *details/name* of `Pending/Future` tasks. Only tasks in Failed, Succeeded, `Running` and `Starting/About to Start` `conditions` are present in the `PipelineRun`. For listing `Pending` tasks, the reference `Pipeline` must be examined using `TaskRef` and any `Tasks` which are present in the reference `Pipeline` but not the `PipelineRun` could be marked as `Pending`. This should be a separate PR/Effort as this would require to bring in the corresponding `Pipeline` *(Firehose)* details in the `PipelineRun` List.

Reply:
But this is also a general problem with tekton and the pipeline run not having enough data in it.
I'm fine if we take this up as a bug.